plugins/destination/s3/client/read.go

```diff
@@ -43,18 +30,5 @@ func (c *Client) Read(ctx context.Context, table *schema.Table, sourceName strin
 		return err
 	}
 	r := bytes.NewReader(writerAtBuffer.Bytes())
-
-	switch c.pluginSpec.Format {
-	case FormatTypeCSV:
-		if err := csv.Read(r, table, sourceName, res); err != nil {
-			return err
-		}
-	case FormatTypeJSON:
-		if err := json.Read(r, table, sourceName, res); err != nil {
-			return err
-		}
-	default:
-		panic("unknown format " + c.pluginSpec.Format)
-	}
-	return nil
+	return c.Client.Read(r, table, sourceName, res)
```

Comment:
can noRotate check be moved to filetype plugin?
that way embedded client func will be enough

Reply:
No, because the NoRotate is only used when the destination doesn't support appending, so it needs to be in the destination package rather than filetype package

Reply:
There're several reasons why I consider this:

- the spec that controls such behavior (has field NoRotate) is in filetypes package
- client is initialized using that spec
- client is also in that filetypes package

It seems natural to move noRotate check to the filetypes client impl then.

Reply:
I think I will just move the NoRotate out of filetypes